daemon/rpcserver.go

```diff
@@ -580,23 +581,39 @@ func (r *rpcServer) WatchAddress(ctx context.Context,
 func (r *rpcServer) NewAddress(ctx context.Context,
 	in *lnrpc.NewAddressRequest) (*lnrpc.NewAddressResponse, error) {
 
-	// Translate the gRPC proto address type to the wallet controller's
-	// available address types.
-	var addrType lnwallet.AddressType
-	switch in.Type {
-	case lnrpc.NewAddressRequest_WITNESS_PUBKEY_HASH:
-		addrType = lnwallet.WitnessPubKey
-	case lnrpc.NewAddressRequest_NESTED_PUBKEY_HASH:
-		addrType = lnwallet.NestedWitnessPubKey
+	if len(in.SubmarineHash) == 0 {
```

Comment:
I would rather put the submarine flow in the if statement because that's the special case, but it's a matter of preference